corehq/apps/user_importer/importer.py

```diff
@@ -860,7 +860,7 @@ def process_new_user(self):
 
             user_invite_loc_id = None
             user_invite_locs_ids = []
-            if self.domain_info.can_assign_locations and cv['location_codes']:
+            if self.domain_info.can_assign_locations and cv['location_codes'] is not None:
```

Comment:
note: This is the primary fix for the bug to be addressed.

This change here makes the logic same for the "new" user as it is done above for "existing user" in function `_modify_existing_user_in_domain`.